src/main/kotlin/org/jetbrains/research/commentupdater/inspection/CodeCommentInspection.kt

```diff
@@ -0,0 +1,116 @@
+package org.jetbrains.research.commentupdater.inspection
+
+import com.intellij.codeInspection.*
+import com.intellij.openapi.vcs.changes.Change
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.vcs.changes.ChangeListManager
+
+import com.intellij.psi.*
+import com.intellij.psi.javadoc.PsiDocComment
+import gr.uom.java.xmi.diff.*
+import org.jetbrains.research.commentupdater.CodeCommentDiffs
+import org.jetbrains.research.commentupdater.JITDetector
+import org.jetbrains.research.commentupdater.processors.CodeCommentTokenizer
+import org.jetbrains.research.commentupdater.processors.MethodChangesExtractor
+import org.jetbrains.research.commentupdater.processors.RefactoringExtractor
+import org.refactoringminer.api.Refactoring
+import org.refactoringminer.api.RefactoringType
+import kotlin.io.path.ExperimentalPathApi
+
+
+class CodeCommentInspection : AbstractBaseJavaLocalInspectionTool() {
+
+    private val LOG: Logger = Logger.getInstance("#org.jetbrains.research.commentupdater.inspection.CodeCommentInspection")
+    val detector = JITDetector()
+
+    var currentFile = ""
+    var currentChanges: Change? = null
+    var currentMethodsRefactorings = hashMapOf<String, MutableList<Refactoring>>()
+
+    val DESCRIPTION_TEMPLATE = "Inconsistent comment found"
+
+    override fun inspectionStarted(session: LocalInspectionToolSession, isOnTheFly: Boolean) {
+        LOG.info("Inspection started")
+
+        // Extract changes
+
+        currentFile = session.file.name
+        val changeListManager = ChangeListManager.getInstance(session.file.project)
+        currentChanges = changeListManager.getChange(session.file.virtualFile)
+
+        // Extract Refactorings
+        currentChanges?.let {
+            val refactorings = RefactoringExtractor.extract(it)
+            currentMethodsRefactorings = RefactoringExtractor.methodsToRefactoringTypes(refactorings)
+        }
+
+        LOG.info("File ${currentFile}, Refactorings: $currentMethodsRefactorings")
+
+        super.inspectionStarted(session, isOnTheFly)
+    }
+
+    override fun inspectionFinished(session: LocalInspectionToolSession, problemsHolder: ProblemsHolder) {
+        LOG.info("Inspection finished")
+        super.inspectionFinished(session, problemsHolder)
+    }
+
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : JavaElementVisitor() {
+
+            override fun visitDocComment(comment: PsiDocComment?) {
+                LOG.info("I am visiting DocComment inside $currentFile")
```

Comment:
Maybe we should combine this message with 66 line? "[CommentUpdater] Processing comment in $currentFile. Comment text: " +  comment.text. And print it on 65 line